### PR TITLE
Fragment Requests

### PR DESCRIPTION
**Pull request by ewz4**
# Fragment Requests

### Summary of changes
Added support for camera and IMU report fragment requests.

### Testing
Tested on Flatsat in several deployment and downlink scenarios

### SFR Changes
Added variables for fragment requesting

### CODE REVIEW

**Comment by dhacks541** [comment]
Latest version of the camera and IMU downlink functionality has been successfully tested in simulated deployment scenario, requesting review for merge.

**Review by Duncan-McD — CHANGES_REQUESTED** [review]
Agreed, looks great aside from rockblock command factory issue

File: src/Control Tasks/RockblockControlTask.cpp [hunk 1]
```diff
@@ -557,16 +557,16 @@ RockblockCommand *RockblockControlTask::commandFactory(RawRockblockCommand raw)
...
 #endif
-        return new FaultOverrideCommand(raw);
-    } else if (op_code <= constants::rockblock::opcodes::sfr_field_opcode_max && op_code >= constants::rockblock::opcodes::sfr_field_opcode_min) {
+        return new CameraFragmentCommand(raw);
+    } else if (op_code == constants::rockblock::opcodes::sfr_field_opcode_imu_fragment_request) {
 #ifdef VERBOSE_RB
-        Serial.println("SFR Override Command");
+        Serial.println("IMU Fragment Request");
 #endif
-        return new SFROverrideCommand(raw);
```
[the remark this answers is not in the document]
Comment (jonathanjma) [reply src/Control Tasks/RockblockControlTask.cpp:569]: fixed

**Comment by dhacks541** [comment]
> Looks good other than that one comment. Also, wanted to confirm all of these changes have been tested with GS in the loop (for example, requesting an IMU fragment from the GS).

And yeah tested with the GS

**Review by ljgreenhill — APPROVED** [review]

**Pull request merged**